hubble/utils/auth.py

```diff
@@ -21,60 +22,37 @@ def get_auth_token():
         return token_from_env if token_from_env else config.get('auth_token')
 
     @staticmethod
-    async def login():
+    async def login(**kwargs):
         api_host = get_base_url()
-
+        auth_info = None
         async with aiohttp.ClientSession(trust_env=True) as session:
-            redirect_url = 'http://localhost:8085'
+            kwargs['redirect_url'] = 'https://hub.jina.ai/authorization-landing'
+            kwargs['provider'] = 'jina-login'
 
             async with session.get(
                 url=urljoin(
                     api_host,
-                    'user.identity.authorize?provider=jina-login&redirectUri=',
-                )
-                + redirect_url
+                    'user.identity.proxiedAuthorize?{}'.format(urlencode(kwargs)),
+                ),
             ) as response:
-                response.raise_for_status()
-                json_response = await response.json()
-                webbrowser.open(json_response['data']['redirectTo'], new=2)
-
-        done = False
-        post_data = None
-
-        class S(BaseHTTPRequestHandler):
-            def _set_response(self):
-                self.send_response(200)
-                self.send_header('Content-type', 'text/html')
-                self.end_headers()
-
-            def do_POST(self):
-                nonlocal done, post_data
-
-                content_length = int(self.headers['Content-Length'])
-                post_data = parse_qs(self.rfile.read(content_length))
-
-                self._set_response()
-                with open(
-                    os.path.join(
-                        os.path.dirname(__file__), 'template', 'logged_in_page.html'
-                    ),
-                    'rb',
-                ) as f:
-                    self.wfile.write(f.read())
-                done = True
-
-            def log_message(self, format, *args):
-                return
-
-        server_address = ('', 8085)
-        with HTTPServer(server_address, S) as httpd:
-            while not done:
-                httpd.handle_request()
+                async for line in response.content:
+                    item = json.loads(line.decode('utf-8'))
+                    event = item['event']
+                    if event == 'redirect':
+                        if in_google_colab():
+                            print('Please open the following link in your browser:')
+                            print(item['data']['redirectTo'])
```

Comment:
Should we use the fancy way to open URL directly?
https://github.com/jina-ai/hubble-client-python/pull/43/commits/fa1fb838e5bae1a781be67246c0e01f6472ceb35
```suggestion
                            from IPython.display import Javascript, display

                            display(
                                Javascript(
                                    f'window.open({json_response["data"]["redirectTo"]});'
                                )
                            )
```